lib/src/over_react_redux/devtools/middleware.dart

```diff
@@ -0,0 +1,158 @@
+    log.onRecord.listen((rec) {
+      if (rec.level == Level.WARNING) {
+        window.console.warn('${log.name} [${rec.level.name}]: ${rec.message}');
+      } else {
+        window.console.log('${log.name} [${rec.level.name}]: ${rec.message}');
+      }
+    });
+    try {
+      devToolsExt = reduxExtConnect();
+    } catch(_) {
+      log.warning(
+        'Unable to connect to the redux dev tools browser extension.\n'
+        'Please install it...\n'
+        'Chrome: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en\n'
+        'Firefox: https://addons.mozilla.org/en-US/firefox/addon/reduxdevtools/\n'
+      );
+    }
+  }
+
+  set store(Store v) {
+    _store = v;
+    devToolsExt.init(_encodeForTransit(store.state));
+    devToolsExt.subscribe(handleEventFromRemote);
+  }
+
+  get store => _store;
+
+  dynamic _encodeForTransit(dynamic content, {bool shouldRethrow = false}) {
+    try {
+      return jsify(jsonDecode(jsonEncode(content)));
+    } catch (_) {
+      log.warning('You must implement a `toJson` method in your state and actions in order to view state changes in the redux dev tools.');
+      if (shouldRethrow) rethrow;
+    }
+  }
+
+  dynamic _encodeActionForTransit(dynamic action) {
+    try {
+      return _encodeForTransit({'type': _getActionType(action), 'payload': action}, shouldRethrow: true);
+    } on Error {
+      return _encodeForTransit({'type': _getActionType(action)});
+    }
+  }
+
+  String _getActionType(dynamic action) {
+    if (action.toString().contains('Instance of')) {
+      return action.runtimeType.toString();
+    }
+    return action.toString();
+  }
+
+  _relay(String type, [Object state, dynamic action, String nextActionId]) {
+    final message = JsBackedMap();
+    message['type'] = type;
+
+    if (state != null) {
+      try {
+        message['payload'] = _encodeForTransit(state, shouldRethrow: true);
+      } catch (_) {
+        message['payload'] = 'Could not encode state. Ensure state is json encodable';
+      }
+    }
+
+    if (type == 'ACTION') {
+      message['action'] = _encodeActionForTransit(action);
+      message['nextActionId'] = nextActionId;
+    } else if (action != null) {
+      message['action'] = action as String;
+    }
+
+    if (message.containsKey('action')) {
+      devToolsExt.send(message['action'], message['payload']);
+    } else {
+      devToolsExt.send(message.jsObject);
+    }
+  }
+
+  void handleEventFromRemote(JsMap jsData) {
+    JsBackedMap data = JsBackedMap.fromJs(jsData);
+    switch (data['type'] as String) {
+      case 'DISPATCH':
+        _handleDispatch(JsBackedMap.fromJs(data['payload']));
+        break;
+      case 'ACTION':
+        _handleRemoteAction(data['action'] as String);
+        break;
+      default:
+        log.warning('Unknown type:' + data['type'].toString());
+    }
+  }
+
+  void _handleDispatch(dynamic action) {
+    if (this._store == null) {
```

Comment:
#nit Unnecessary use of this here and a couple other places: https://dart.dev/guides/language/effective-dart/usage#dont-use-this-when-not-needed-to-avoid-shadowing